src/wrapper.jl

```diff
@@ -106,22 +214,952 @@ const FT_PURGE_TX = 2
+end
+
+
+
+"""
+    FT_GetDeviceInfoDetail(dwIndex)
+
+Wrapper for D2XX library function `FT_GetDeviceInfoDetail`. 
+
+See D2XX Programmer's Guide (FT_000071) for more information.
+
+# Arguments
+ - `dwIndex`: Index of entry in the device info list.
+
+# Example
+```julia-repl
+julia> numdevs = FT_CreateDeviceInfoList()
+0x00000004
+
+julia> idx, flags, typ, id, locid, serialnumber, description, fthandle = FT_GetDeviceInfoDetail(0) # zero indexed
+(0, 0x00000002, 0x00000007, 0x04036011, 0x00000000, "FT3AD2HCD", "USB <-> Serial Converter D", FT_HANDLE(Ptr{Nothing} @0x0000000000000000))
+```
+"""
+function FT_GetDeviceInfoDetail(dwIndex)
+  lpdwFlags, lpdwType  = Ref{DWORD}(), Ref{DWORD}()
+  lpdwID,    lpdwLocId = Ref{DWORD}(), Ref{DWORD}()
+  pcSerialNumber = pointer(Vector{Cchar}(undef, 16))
+  pcDescription  = pointer(Vector{Cchar}(undef, 64))
+  ftHandle = FT_HANDLE()
+  
+  status = ccall(cfunc[:FT_GetDeviceInfoDetail], cdecl, FT_STATUS, 
+  (DWORD,   Ref{DWORD}, Ref{DWORD}, Ref{DWORD}, Ref{DWORD}, Cstring,        Cstring,       Ref{FT_HANDLE}),
+   dwIndex, lpdwFlags,  lpdwType,   lpdwID,     lpdwLocId,  pcSerialNumber, pcDescription, ftHandle)
+  
+  FT_STATUS_ENUM(status) == FT_OK || throw(FT_STATUS_ENUM(status))
+  dwIndex[], lpdwFlags[], lpdwType[], lpdwID[], lpdwLocId[], unsafe_string(pcSerialNumber), unsafe_string(pcDescription), ftHandle
+end
+
+
+
+"""
+    FT_ListDevices(pvArg1, pvArg2, dwFlags)
+
+**NOT FULLY FUNCTIONAL: NOT RECOMMENDED FOR USE**.
+
+Wrapper for D2XX library function `FT_ListDevices`.
+
+See D2XX Programmer's Guide (FT_000071) for more information.
+
+# Arguments
+ - `pvArg1`: Depends on dwFlags.
+ - `pvArg2`: Depends on dwFlags.
+- `dwFlags`: Flag which determines format of returned information.
+
+E.g. call with `pvArg1 = Ref{UInt32}()` and/or `pvArg2 = Ref{UInt32}()` for 
+cases where `pvArg1` and/or `pvArg2` return or are given DWORD information.
+
+# Examples
+
+1. Get number of devices...
+```julia-repl
+
+julia> numdevs = Ref{UInt32}();
+
+julia> FT_ListDevices(numdevs, Ref{UInt32}(), FT_LIST_NUMBER_ONLY)
+
+julia> numdevs[]
+0x00000004
+```
+
+2. Get serial number of first device... *NOT CURRENTLY WORKING*
+```julia-repl
+
+julia> devidx = Ref{UInt32}(0)
+Base.RefValue{UInt32}(0x00000000)
+
+julia> buffer = pointer(Vector{Cchar}(undef, 64))
+Ptr{Int8} @0x0000000004f2e530
+
+julia> FT_ListDevices(devidx, buffer, FT_LIST_BY_INDEX|FT_OPEN_BY_SERIAL_NUMBER)
+ERROR: FT_ListDevices wrapper does not yet flags other than FT_LIST_NUMBER_ONLY.
+Stacktrace:
+...
+
+```
+"""
+function FT_ListDevices(pvArg1, pvArg2, dwFlags)
+  dwFlags == FT_LIST_NUMBER_ONLY || throw(ErrorException("FT_ListDevices wrapper does not yet flags other than FT_LIST_NUMBER_ONLY."))
+  flagsarg = DWORD(dwFlags)
+  status = ccall(cfunc[:FT_ListDevices], cdecl, FT_STATUS, 
+                 (Ptr{Cvoid}, Ptr{Cvoid}, DWORD),
+                  pvArg1,     pvArg2,     dwFlags)
+  FT_STATUS_ENUM(status) == FT_OK || throw(FT_STATUS_ENUM(status))
```

Comment:
This line is repeated everywhere make a function that takes status e..g `_checkstatus(status) = ....`

Reply:
resolved in 3a7e05f49eca09bf4edc8c1d05e19eb82e93ffaa